src/components/organisms/SelectionControlsInput/SelectionControlsInput.tsx

```diff
@@ -0,0 +1,156 @@
+import { ReactElement, forwardRef, useRef, useState } from 'react'
+import classNames from 'classnames'
+import { Field, Label } from '@radix-ui/react-form'
+import { FieldError } from 'react-hook-form'
+import InputError from 'components/atoms/InputError/InputError'
+import BaseButton from 'components/atoms/BaseButton/BaseButton'
+import { ReactComponent as DropdownArrow } from 'assets/icons/dropdown.svg'
+import { useClickAway } from 'ahooks'
+import DropdownContent from 'components/organisms/DropdownContent/DropdownContent'
+
+import classes from './styles.module.scss'
+import { filter, includes, join, map } from 'lodash'
+
+export enum DropdownSizeTypes {
+  L = 'l',
+  M = 'm',
+  S = 's',
+}
+
+export interface SelectionControlsInputProps {
+  name: string
+  className?: string
+  error?: FieldError
+  label?: JSX.Element | string
+  ariaLabel: string
+  value?: string
+  options: {
+    label: string
+    value: string
+  }[]
+  onChange: (value: string | string[]) => void
+  disabled?: boolean
+  defaultLabel?: string
+  multiple?: boolean
+  helperText?: string
+  buttons?: boolean
+  cancelButtonLabel?: string
+  proceedButtonLabel?: string
+  searchInput?: ReactElement
+  dropdownSize?: DropdownSizeTypes
+  tags?: boolean
+}
+
+const SelectionControlsInput = forwardRef<
+  HTMLInputElement,
+  SelectionControlsInputProps
+>(function SelectionControlsInput(
+  {
+    label,
+    name,
+    ariaLabel,
+    className,
+    value,
+    error,
+    options,
+    onChange,
+    disabled,
+    defaultLabel,
+    multiple = false,
+    helperText,
+    buttons = false,
+    cancelButtonLabel,
+    proceedButtonLabel,
+    searchInput,
+    dropdownSize,
+    tags = false,
+  },
+  ref
```

Comment:
**[blocker]**

ref doesn't seem to be used, so this forwardRef is redundant. Although before removing it I would suggest going over react-hook-form docs, so you understand what it uses ref for (It's not necessary, but then we have to handle some of the same things manually)

Reply:
**[blocker]**

ref still seems to be in the wrong place. Currently when selection input is the first input in the form and I get an error on it after submitting, it won't be focused. Extra note here is that I wasn't exactly right, when I said that each html element can be focused. It's still basically true, but it's not true by default. Would suggest reading a bit more about how focus works:

https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/focus


In case of selection we want to focus on an element that would allow for the opening of the dropdown with the press of 1 button on keyboard afterwards.